src/module.c

```diff
@@ -5753,12 +5751,17 @@ robj **moduleCreateArgvFromUserFormat(const char *cmdname, const char *fmt, int
  *     * `0` -- Return the reply in auto mode, i.e. the reply format will be the
  *              same as the client attached to the given RedisModuleCtx. This will
  *              probably used when you want to pass the reply directly to the client.
- *     * `C` -- Check if command can be executed according to ACL rules.
- *              Uses the user specified by RM_SetContextModuleUser, if set, otherwise
- *              if the context has an attached client, will use the user associated
- *              with the client. If neither are available, will return an error.
- *              In addition, when defined, will enforce ACL checking within scripts
- *              as well.
+ *     * `C` -- Run a command as the user attached to the context.
+ *              User is either attached automatically via the client that directly
+ *              issued the command and created the context or via RM_SetContextModuleUser.
+ *              If the context is not directly created by an issued command (such as a
+ *              background context and no user was set on it via RM_SetContextModuleUser,
+ *              RM_Call will fail.
+ *              Checks if the command can be executed according to ACL rules and causes
+ *              the command to run as the determined user, so that any future user
+ *              dependent activity, such as ACL checks within scripts will proceed as
+ *              expected.
+ *              Otherwise, the command will run as the Redis unrestricted user.
```

Comment:
I was leaning towards using the user set via RM_SetContextModuleUser even without "C", but now that you've re-purposed "C", it actually makes sense that way too.

However, this is a behavior change causing scripts that used to work to stop working when redis is upgraded.
but since it only happens with "C", and its purpose was to check ACL, which scripts actually bypassed, i think we can consider that a bugfix.
@MeirShpilraien @yossigo please ack.

@sjpotter please make sure to reflect that in the PR's top comment.
also, i see you opened #11231, i'm not sure why (it covers the same purpose that you fixed here, right)?
shall we set it to be closed when this one is merged?

p.s. how is that in terms of symmetry with our intents in #11161 (a flag that tells RM_Call to affect commands in scripts the same way it affect normal commands)

Reply:
yes, I opened 11231 to make this a discussion of this being a bugfix or not.  i.e. that if we needed 'C' semantics to stay the way they are, perhaps we deprecate 'C' (i.e. we still support it, but note that it doesn't do what one might expect for scripts) and add a 'U' that is much more clear of what happens in this case.  (as I describe in the comment)

I can imagine it also possibly being useful for other modules (not sure how they would use, but my imagination says it might be possible), as a way to "sandbox" modules.  i.e. a module could restrict itself by always running as a more restricted user.  Unsure this is really necessary, but perhaps others could take it and run with it.